### PR TITLE
Adding Halts. Other 1860 tile rendering fixes.

### PR DESCRIPTION
**Pull request by roseundy**
Created new tile part: Halt, which is a subclass of Town.
Changed 1860 tile definition to use Halts.
Changed rendering to handle Halts.
Also made other rendering tweaks/fixes so 1860 tiles display correctly.

New:

### CODE REVIEW

File: assets/app/view/game/part/town_dot.rb [hunk 1]
```diff
@@ -69,32 +69,54 @@ def preferred_render_locations
...
 
           displacement = @edge ? REVENUE_EDGE_DISPLACEMENT : REVENUE_DISPLACEMENT
 
           increment_weight_for_regions(REVENUE_REGIONS[layout])
-          h(:g, { attrs: { transform: "rotate(#{angle})" } }, [
-              h(:g, { attrs: { transform: "translate(#{displacement} 0) rotate(#{-angle})" } }, [
-                  h(Part::SingleRevenue,
-                    revenue: revenue,
-                    transform: rotation_for_layout),
+          if @town.halt?
```
Comment (tobymao) [inline assets/app/view/game/part/town_dot.rb:92]: seems like there's a lot of duplicate code here that could be combined

File: assets/app/view/game/part/town_rect.rb [hunk 2]
```diff
@@ -121,15 +122,25 @@ def render_revenue
...
-            h(:g, { attrs: { transform: "rotate(#{angle})" } }, [
-              h(:g, { attrs: { transform: "translate(#{displacement} 0) #{rotation_for_layout}" } }, [
-                h(SingleRevenue,
-                  revenue: revenue,
-                  transform: "rotate(#{-angle})"),
-              ]),
-            ]),
-          ])
+          if @town.halt?
+            h(:g, { key: "#{@town.id}-r", attrs: { transform: "translate(#{x.round(2)} #{y.round(2)})" } }, [
```
Comment (tobymao) [inline assets/app/view/game/part/town_rect.rb:126]: same here

File: lib/engine/part/halt.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+require_relative 'town'
+
+module Engine
+  module Part
+    class Halt < Town
+      attr_reader :symbol
+
+      def initialize(symbol, **opts)
+        @revenue = parse_revenue('0', opts[:format])
```
Comment (tobymao) [inline lib/engine/part/halt.rb:11]: seems luke duplicate code with town, can you call super and override?

File: assets/app/view/game/part/town_dot.rb (hunk 1 above)
Comment (roseundy) [reply assets/app/view/game/part/town_dot.rb:92]: Ok

File: assets/app/view/game/part/town_rect.rb (hunk 2 above)
Comment (roseundy) [reply assets/app/view/game/part/town_rect.rb:126]: Ok

File: lib/engine/part/halt.rb (hunk 3 above)
Comment (roseundy) [reply lib/engine/part/halt.rb:11]: Ok, need to learn how to change arguments to the super

**Comment by roseundy** [comment]
Made the suggested changes.

**Review by tobymao — APPROVED** [review]

**Pull request merged**